### PR TITLE
Parameter fix when using PBS options

### PR DESCRIPTION
**Pull request by milljm**
While using PBS options, custom test parameters were not being correctly
populated in the pbs_template file. Arguments such as --timing for example
would not appear in the PBS cluster file.

Refs #8778

### CODE REVIEW

**Review by permcody — CHANGES_REQUESTED** [review]
Since you are refactoring all this code, why not just get the actual command from the getCommand() method. That'll take care of most of this logic. Sure you'll still need to do a few extra things for PBS but start with the fact that the command to execute and all that logic is already handled for you.

File: python/TestHarness/testers/RunApp.py [hunk 1]
```diff
@@ -177,93 +177,69 @@ def getCommand(self, options):
...
 
-        return command
 
-    def getPBSCommand(self, options):
-        if options.parallel == None:
-            default_ncpus = 1
-        else:
-            default_ncpus = options.parallel
+            # Squash cli_args list. Needs to happen before we attempt to replace template with valid parameters.
+            specs['cli_args'] = '"' + ' '.join(specs['cli_args']) + '"'
```
Comment (permcody) [inline python/TestHarness/testers/RunApp.py]: Getting closer. However we are still doing one or two things in both major branches here. Building the actual command cli_args appears to be one of those things that's duplicated. Can this be captured from the other branch (i.e. factor it out and do cli_args in one place only).

**Review by permcody — CHANGES_REQUESTED** [review]

File: python/TestHarness/testers/RunApp.py (hunk 1 above)
Comment (milljm) [inline python/TestHarness/testers/RunApp.py]: Sorry, I am not following. I need to squash this into itself, and encapsulate it in quotes so when the replace routine occurs:
```python
            # Do all of the replacements for the valid parameters
            for spec in specs.valid_keys():
                if spec in specs.substitute:
                    specs[spec] = specs.substitute[spec].replace(spec.upper(), specs[spec])
                content = content.replace('<' + spec.upper() + '>', str(specs[spec]))
```
It writes the correct thing in the cluster file (not a list):
```bash
[Jobs]
  [./xda_re_05]
    ...
    cli_args = "--error --no-gdb-backtrace --timing"
  [../]
[]
```
I suppose I can remove this all together and then fix it in the cluster_launcher.py script instead?

**Comment by permcody** [comment]
Closing due to inactivity.

**Pull request closed without merging**